.github/workflows/linter.yml

```diff
@@ -0,0 +1,42 @@
+name: Linter
+
+on:
+  push:
+    paths:
+      - '**.py'
+  pull_request:
+    paths:
+      - '**.py'
+
+jobs:
+  flake8_py3:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.7
+
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('requirements/*.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Install project requirements
+        run: |
+          python -m pip install -U -r requirements/base.txt
+          python -m pip install -U -r requirements/development.txt
+          python -m pip install --no-deps -U -r requirements/embedded.txt -t ext_libs
```

Comment:
Just for curiosity, why do you need these pip install ?
AFAIK, the action below won't use it, no?

Reply:
To avoid to get flake8 reporting import errors, I need to install the dependencies. I could justinstall the required ones (embedded), but in my base I've got setuptools and wheel which are useful for improve install and caching steps; and in me development I've got flake8 so it's more like a shortcut.

Reply:
Afaik is not complaining if the underlying library is not available.
Or did you activate an extension ? (Pylint is doing it)

For instance, every import using `from qgis.` are not failing, correct? (and there isn't any qgis installed)

Reply:
Hmmm you're right, I've been using flake8 and pylint in combination for a long time. Maybe it's time to make things lighter.

Reply:
Ok, thanks for checking.
I'm not sure why flake8 is not happy now.

My question was related to the 3 lines of `pip install` as well (L33, L34)

Anyway, feel free to skip if needed. It was just for curiosity.

Reply:
> I'm not sure why flake8 is not happy now.

Because there were some errors to spot.

> My question was related to the 3 lines of pip install as well (L33, L34)

This way, I keep advantage from my version definition of flake8 and cache.

Reply:
Finally, I did not like how the external action works. I prefer to apply my flake8 configuration.

Reply:
> Finally, I did not like how the external action works. I prefer to apply my flake8 configuration.

I totally agree. I also switch using a plain flake8 command instead of a action ;-)

Thanks for your explanations